client/rest-high-level/src/test/java/org/elasticsearch/client/dataframe/transforms/hlrc/DataFrameTransformStatsTests.java

```diff
@@ -58,48 +46,65 @@ public static NodeAttributes fromHlrc(org.elasticsearch.client.dataframe.transfo
             attributes.getAttributes());
     }
 
-    @Override
-    public org.elasticsearch.client.dataframe.transforms.DataFrameTransformState doHlrcParseInstance(XContentParser parser)
-            throws IOException {
-        return org.elasticsearch.client.dataframe.transforms.DataFrameTransformState.fromXContent(parser);
+    public static DataFrameTransformStats
+        fromHlrc(org.elasticsearch.client.dataframe.transforms.DataFrameTransformStats instance) {
+
+        return new DataFrameTransformStats(instance.getId(),
+            DataFrameTransformTaskState.fromString(instance.getTaskState().value()),
+            instance.getReason(),
+            fromHlrc(instance.getNode()),
+            DataFrameIndexerTransformStatsTests.fromHlrc(instance.getIndexerStats()),
+            DataFrameTransformCheckpointingInfoTests.fromHlrc(instance.getCheckpointingInfo()));
     }
 
     @Override
-    public DataFrameTransformState convertHlrcToInternal(org.elasticsearch.client.dataframe.transforms.DataFrameTransformState instance) {
-        return fromHlrc(instance);
+    public org.elasticsearch.client.dataframe.transforms.DataFrameTransformStats doHlrcParseInstance(XContentParser parser)
+            throws IOException {
+        return org.elasticsearch.client.dataframe.transforms.DataFrameTransformStats.fromXContent(parser);
     }
 
     @Override
-    protected DataFrameTransformState createTestInstance() {
-        return randomDataFrameTransformState();
+    public DataFrameTransformStats convertHlrcToInternal(
+        org.elasticsearch.client.dataframe.transforms.DataFrameTransformStats instance) {
+        return new DataFrameTransformStats(instance.getId(),
+                DataFrameTransformTaskState.fromString(instance.getTaskState().value()),
+                instance.getReason(),
+                fromHlrc(instance.getNode()),
+                DataFrameIndexerTransformStatsTests.fromHlrc(instance.getIndexerStats()),
+                DataFrameTransformCheckpointingInfoTests.fromHlrc(instance.getCheckpointingInfo()));
+    }
+
+    public static DataFrameTransformStats randomDataFrameTransformStats() {
+        return new DataFrameTransformStats(randomAlphaOfLength(10),
+            randomFrom(DataFrameTransformTaskState.values()),
+            randomBoolean() ? null : randomAlphaOfLength(100),
+            randomBoolean() ? null : randomNodeAttributes(),
+            // On the server side the stats has transform ID "_all" when embedded in another doc
+            // TODO: change this so that the outer document sets the correct transform ID during parsing
+            // It's very confusing and could cause subtle errors that the inner object has a surprising ID
```

Comment:
just for the record: the id in stats is a refactoring leftover, it was forgotten to be removed, stats was stored as individual document before state and stats have been combined, see discussion above. So the todo should be "remove the id"